lib/internal/errors.js

```diff
@@ -622,6 +622,45 @@ function addNumericalSeparator(val) {
   return `${val.slice(0, i)}${res}`;
 }
 
+// Used to enhance the stack that will be picked up by the inspector
+const kEnhanceStackBeforeInspector = Symbol('kEnhanceStackBeforeInspector');
+
+// These are supposed to be called only on fatal exceptions before
+// the process exits.
+const fatalExceptionStackEnhancers = {
+  beforeInspector(error) {
+    if (typeof error[kEnhanceStackBeforeInspector] !== 'function') {
+      return error.stack;
+    }
+
+    try {
+      // Set the error.stack here so it gets picked up by the
+      // inspector.
+      error.stack = error[kEnhanceStackBeforeInspector]();
```

Comment:
Ideally we probably should not even have to do this, instead we can just pass a string to some API and leave the stack intact, because patching `error.stack` over and over is error-prone...but there does not seem to be a hook to customize the stack trace in the console message logged by the inspector.

Reply:
Yeah and error.stack might not be writeable, (should be fine in this case)